### PR TITLE
 feat: add `Unexpander` for delaborating without importing `Lean`, use for simple notations

### PR DESCRIPTION
**Pull request by Kha**

### CODE REVIEW

File: src/Init/Notation.lean [hunk 1]
```diff
@@ -18,25 +18,29 @@ infixl:70 " / "  => Div.div
 infixl:70 " % "  => Mod.mod
 infixl:70 " %ₙ " => ModN.modn
 infixr:80 " ^ "  => Pow.pow
+prefix:100 "-"   => Neg.neg
 
-infix:50 " ≤ "  => HasLessEq.LessEq
+-- declare ASCII alternatives first so that the latter Unicode unexpander wins
```
Comment (Kha) [inline src/Init/Notation.lean:23]: Issue: How to restore the `pp.unicode` behavior? When synthesizing the unexpander, we could reasonably check for Unicode characters in the tokens and emit a check for said option in that case. Notations without ASCII alternatives would be printed as ordinary implications in that case, but perhaps that's actually the more reasonable behavior.

File: src/Lean/Parser/Term.lean [hunk 2]
```diff
@@ -170,11 +170,6 @@ def matchAltsWhereDecls := parser! matchAlts false >> optional whereDecls
 @[builtinTermParser] def ensureTypeOf       := parser! "ensureTypeOf! " >> termParser maxPrec >> strLit >> termParser
 @[builtinTermParser] def ensureExpectedType := parser! "ensureExpectedType! " >> strLit >> termParser maxPrec
 
-@[builtinTermParser] def not    := parser! "¬" >> termParser 40
-@[builtinTermParser] def bnot   := parser! "!" >> termParser 40
--- symbol precedence should be higher, but must match the one of `sub` below
```
Comment (Kha) [inline src/Lean/Parser/Term.lean:175]: @leodemoura Good riddance :)

File: src/Init/Notation.lean (hunk 1 above)
Comment (leodemoura) [reply src/Init/Notation.lean:23]: I agree.

**Comment by leodemoura** [comment]
@Kha I merged manually. The `HMul` refactor was creating more conflict.

**Pull request closed without merging**